addons/mrp/models/mrp_bom_report.py

```diff
@@ -0,0 +1,139 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models
+
+
+class MrpBomReport(models.TransientModel):
+    _name = 'mrp.bom.report'
+    _description = "Mrp Bom Report"
+
+    def _get_price(self, bom, line_qty):
+        price = 0
+        for line in bom.bom_line_ids:
+            if line.child_bom_id:
+                sub_price = self._get_price(line.child_bom_id, line.product_qty)
+                price += sub_price * line.product_qty
+            else:
+                prod_qty = (line.product_qty * line_qty) / bom.product_qty
+                price += (line.product_id.uom_id._compute_price(line.product_id.standard_price, line.product_uom_id) * prod_qty) / line_qty
+        return price
+
+    @api.model
+    def get_lines(self, bom_id=False, line_qty=False, level=False):
+        context = self.env.context or {}
+        data = []
+        bom = self.env['mrp.bom'].browse(bom_id or context.get('active_id'))
+
+        if bom:
+
+            products = bom.product_id or bom.product_tmpl_id.product_variant_ids or bom.product_tmpl_id
+            for product in products:
+                lines = {}
+                components = []
+                lines.update({
+                    'bom': bom,
+                    'bom_prod_name': bom.product_id.display_name or bom.product_tmpl_id.display_name or '',
+                    'currency': self.env.user.company_id.currency_id,
+                    'total': 0.0,
+                    'attribute': product.attribute_value_ids,
+                })
+                bom_lines = bom.bom_line_ids
+                if product.attribute_value_ids:
+                    bom_lines = bom.bom_line_ids.filtered(lambda l: product.attribute_value_ids in l.attribute_value_ids)
```

Comment:
Dear 

I will better to use this methode `_skip_bom_line(self, product)` to show or not the line : https://github.com/odoo/odoo/blob/10.0/addons/mrp/models/mrp_bom.py#L268

If this methode `_skip_bom_line(self, product)` is overpassed to explose Bom in Product Order (case in my company), the cost structure will be the same.